src/components/FileTreeNode/styles.module.scss

```diff
@@ -40,6 +40,10 @@
   width: 100%;
 }
 
+.expandedFolderIcon {
+  margin-right: -0.125em;
```

Comment:
This still seems fragile because it relies on the width disparity that exists within the FontAwesome library. If they were to, say, fix this bug by making all of their icon widths uniform, it would break our layout.

What if we just controlled the width? Here is an example that sets an explicit width for all icons and aligns content to the right so that they never jump. What do you think?

```diff
diff --git a/src/components/FileTreeNode/index.tsx b/src/components/FileTreeNode/index.tsx
index 033c585..351727f 100644
--- a/src/components/FileTreeNode/index.tsx
+++ b/src/components/FileTreeNode/index.tsx
@@ -246,6 +246,8 @@ export class FileTreeNodeBase<TreeNodeType> extends React.Component<Props> {
       nodeTitle = gettext(`${nodeTitle} (added)`);
     }
 
+    const folderIconProps = { className: styles.folderIcon };
+
     return (
       <React.Fragment>
         <ListGroup.Item as={ItemElement} {...listGroupItemProps}>
@@ -258,9 +260,12 @@ export class FileTreeNodeBase<TreeNodeType> extends React.Component<Props> {
             style={{ paddingLeft: `${adjustedLevel * 10}px` }}
           >
             {isFolder ? (
-              <FontAwesomeIcon icon={isExpanded ? 'folder-open' : 'folder'} />
+              <FontAwesomeIcon
+                {...folderIconProps}
+                icon={isExpanded ? 'folder-open' : 'folder'}
+              />
             ) : (
-              <FontAwesomeIcon icon="file" />
+              <FontAwesomeIcon {...folderIconProps} icon="file" />
             )}
             <span title={nodeTitle} className={styles.nodeName}>
               {node.name}
diff --git a/src/components/FileTreeNode/styles.module.scss b/src/components/FileTreeNode/styles.module.scss
index 7c11b25..0ef0f6d 100644
--- a/src/components/FileTreeNode/styles.module.scss
+++ b/src/components/FileTreeNode/styles.module.scss
@@ -32,6 +32,15 @@
   line-height: 1;
   padding: 0.6rem 1.25rem 0.4rem 1.25rem;
   width: 100%;
+
+  // This is nested for specificity.
+  svg.folderIcon {
+    display: flex;
+    flex-shrink: 0;
+    justify-content: flex-end;
+    // Control the width of all icons to prevent the UI from jumping.
+    width: 1.1em;
+  }
 }
 
 .nodeName,
```

Reply:
@kumar303 Thanks for offering this. But one problem is: [ the `svg` element looks not support  `justify-content` ](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute). The icon is not justified to the right as shown below.
![image](https://user-images.githubusercontent.com/4984681/62559957-44e4c200-b839-11e9-9da0-f6c22b53ca9f.png)

I moved the icon into an span element, and it works now :)
I did not justify the icon to `flex-end`, since it makes the folder icon jumping as the first one in  https://github.com/mozilla/addons-code-manager/pull/958#discussion_r308794507 .